daemon/api.go

```diff
@@ -577,27 +578,53 @@ func waitChange(chg *state.Change) error {
 	return chg.Err()
 }
 
+func ensureUbuntuCore(st *state.State, chg *state.Change, other *state.TaskSet) error {
+	var ss snapstate.SnapState
+	var lastCoreTask *state.Task
+
+	ubuntuCore := "ubuntu-core"
+	err := snapstateGet(st, ubuntuCore, &ss)
+	if err != state.ErrNoState {
+		return nil
+	}
+
+	ts, err := snapstateInstall(st, ubuntuCore, "stable", 0)
+	if err != nil {
+		return err
+	}
+	chg.AddAll(ts)
+
+	tl := ts.Tasks()
+	lastCoreTask = tl[len(tl)-1]
+	other.WaitFor(lastCoreTask)
```

Comment:
The order of tasks in a change or inside a taskset is only meaningful for presentation, but is not necessarily indicative of execution ordering.

I think we want something along these lines instead:
```
state.NewTaskSet(chg.Tasks()...).WaitFor(ts)
chg.AddAll(ts)
```